lib/Dialect/ESI/runtime/CMakeLists.txt

```diff
@@ -143,9 +164,11 @@ target_link_libraries(ESICppRuntime PUBLIC
   nlohmann_json::nlohmann_json
 )
 if(NOT MSVC)
-  target_link_libraries(ESICppRuntime PRIVATE
-    dl
-  )
+  if(NOT ESI_STATIC_RUNTIME)
+    target_link_libraries(ESICppRuntime PRIVATE
```

Comment:
This is for loading other libraries (backends), so you still need it if we're compiling ESI Runtime as a static library.